Intermediate amp

nasearle opened the pull request:
Update the code with a solution to the intermediate course. Re-org code into one solution folder for each course.

nasearle committed:
update(all): update basic solution + amp-accordian

This commit combines the solutions from labs 1 and 2 from the first course (basic amp) into one 'basic-solution'. It also updates the code to use the css from the Simple Article AMPStart template. TODO - update lab instructions to match new look.

This commit adds an intermediate solution folder. The code is currently the same as the basic-solution code, but with an amp-accordian component added to the sidebar. TODO - build out the intermediate solution with the rest of the components in the training outline

nasearle committed:
move figcaption on hero image to right side

nasearle committed:
add styles folder to basic-solution

nasearle committed:
added amp-social-share and amp-form

nasearle committed:
amp-form working in intermediate

nasearle committed:
feat(intermediate): amp-lightbox on form submit

nasearle committed:
feat(intermediate): amp selector working

nasearle committed:
amp-bind working in carousel

nasearle committed:
separated amp-lightbox from form

nasearle committed:
refactor(all): update styles

This commit moves the code away from the ampstart CSS library and updates the styles to be in the more traditional format of one or two classes per element.

nasearle committed:
removed unnecessary styles

nasearle committed:
ordered css rules

nasearle committed:
removed references to ampstart in css

nasearle committed:
updated styles in basic solution

morsssss commented on intermediate-solution/index.html:
```diff
@@ -0,0 +1,500 @@
+
+      .sidebar .dropdown-item,.sidebar .dropdown header,.sidebar .nav-item {
+        margin: 0 0 2rem;
+      }
```
Can you do something to make it clear that these are submenu items?  Indent, change color a bit, etc?

I started with the same sample code [here](http://delsolquartet.com/angel-island-live.html) and kind of like how it turned out...

nasearle committed:
remove lightbox, add lightbox-gallery to carousel

nasearle committed:
feature(index.html): added amp-accordion to page

nasearle committed:
update(intermediate-solution): update styles

This commit makes the following changes:
- reduces the size of the h2 font so that the heading in the headerbar is smaller (removes the media queries that change heading size based on window width)
- removes indentation from text paragraphs and left-aligns all elements after the sidebar
- centers the images in the thumbnail carousel
- adds a background color to the sidebar
- makes the background images in amp-social-share round instead of square
- makes the width of the hero image align with the rest of the content
- moves the subscribe form outside the amp-accordion
- puts the subscribe form inside a card
- adds dummy text inside last accordion section
- orders CSS rules to match the order of elements on the page
- alphabetizes property names in CSS

It also updates the styles in main.css and removes the unused ampstart CSS library files, page.css and page.min.css.

nasearle committed:
feat(index): add lightbox to first image

nasearle committed:
(index.html): replace amp-bind with on:taps in carousel

nasearle committed:
(index): moved subscribe heading out of accordion

morsssss committed:
Margin for main accordion container

morsssss committed:
Merge branch 'intermediate-amp' of https://github.com/google-developer-training/amp-ilt into intermediate-amp

nasearle committed:
added on:slideChange that selects corres. thumbnail image

nasearle committed:
add space between sections and below img

nasearle committed:
indent submenu items in sidebar

morsssss committed:
Added cute text for last element of accordion
Added padding for expanded accordion elements

morsssss committed:
Pull changes, forgetting merge

morsssss committed:
New, clever text and minor accordion CSS fix

nasearle committed:
update(all): update project and basic solution

This commit updates the project and basic solution folders to match the new code and format of the course. In particular, this commit:
- removes the unused AMPstart CSS file, page.css
- removes foo.html
- updates main.css to match the styles in index.html
- adds the thumbnail images, main.css, package.json, package-lock.json, and server.js to project and basic-solution

nasearle committed:
update(basic): updated to match intermediate

This commit updates the styles and structure of basic-solution/index.html to match intermediate-solution/index.html

nasearle committed:
removed amp-selector styles from project/basic

nasearle committed:
feat(all): some changes, fixes, and cleanups

This commit makes several updates to the code. There are two structural changes:
- removes amp-accordion from the main page
- moves the image carousel above the amp-youtube component

This commit makes the following code fixes and cleanups:
- removes the extra indentation and padding-right from the page text
- removes any CSS selectors containing i-amphtml (illegal in AMP)
- removes the small paragraph below the YouTube video
- makes the AMP JS library the second child of <head>
- orders the links to the AMP libraries to match the order in the lab
- replaces some CSS class selectors with tag selectors to simplify the markup
- removes some unused CSS rules
- spaces things out and resizes so that everything fits nicely

nasearle committed:
update(server): added domain so start log is clear

nasearle committed:
update(all): swap amp-sidebar and amp-social-share

This commit removes amp-sidebar from the basic-solution and replaces it with the amp-social-share component. It also reorders the amp component libraries in the intermediate-solution to match the order that they appear in the lab instructions.

nasearle committed:
feat(int-solution): add on slideChange to carousel

This commit adds an on attribute to the carousel to select the corresponding image in the amp-select component when the slide is changed

nasearle committed:
update(int-solution): reordered amp libs to match instructions

nasearle commented:
@DavidScales could you do a quick review to make sure nothing is terribly wrong before I merge?

git clone
cd amp-ilt
git checkout intermediate-amp
cd intermediate-solution
npm install
node server.js

and then just click around and make sure all the amp components work and quickly look through index.html and server.js? And maybe do a quick check of the same files in basic-solution as well? Thanks

DavidScales commented:
Works fine & looks great overall. Only minor nitpicks
* package-lock.json updates when I install
* maybe a link to published content in README
* maybe "not an official Google product" in README
* server.js & styles are missing copyright comment
* copyright comment is outdated in others (2016 instead of 2018)
* CSS could be linted, but probably not a big deal if its tied to instructions
* Lighthouse claims that the canonical URL is invalid
* Lighthouse says the input text font is too small for accessibility

DavidScales commented:
* and update README content

nasearle committed:
update(all): minor fixes

This commit performs the following fixes and updates:
- updates package-lock.json in intermediate solution to match the latest npm install
- updates the readme
- adds copyright comments to main.css and server.js
- updates copyright to 2018 in index.html
- increases the size of the input labels so lighthouse is satisfied

nasearle committed:
update(index.html) update canonical link to absolute

nasearle commented:
Latest commits fix everything except the CSS linting. The CSS matches the AMPStart CSS styles and we want to keep everything consistent.

TODO: update readme with link to instructions when those are published

DavidScales reviewed (commented):
oh yeah... LGTM
